pyglmnet/pyglmnet.py

```diff
@@ -714,20 +716,30 @@ def fit(self, X, y):
             ActiveSet = np.ones(n_features + 1)     # init active set
             z = beta[0] + np.dot(X, beta[1:])       # cache z
 
-        # Initialize loss accumulators
-        L, DL = list(), list()
+        # Iterative updates
         for t in range(0, self.max_iter):
             if self.solver == 'batch-gradient':
                 grad = _grad_L2loss(self.distr,
                                     alpha, self.Tau,
                                     reg_lambda, X, y, self.eta,
                                     beta)
-
+                if t > 1:
+                    if np.linalg.norm(grad) / np.linalg.norm(beta) < tol / lr:
```

Comment:
@pavanramkumar couldn't this be done using a single `if` statement? According to zen of python:

> Flat is better than nested

Reply:
hmm, we get computational savings by not having to compute gradients and parameter norms in the first iteration

Reply:
see https://github.com/glm-tools/pyglmnet/pull/252#issuecomment-419670231